packages/metamask/src/index.ts

```diff
@@ -65,6 +65,7 @@ export class MetaMask extends Connector {
         this.provider.on('disconnect', (error: ProviderRpcError): void => {
           this.actions.resetState()
           this.onError?.(error)
```

Comment:
Did you intentionally `setError` after `onError`? Does order matter here?

Reply:
order does not matter here. onError is set by the external package using web 3 react to do something, in the case of uniswap interface, it is just debug logging to console.